src/utils/permaLinkService.js

```diff
@@ -0,0 +1,11 @@
+export const setPermalinkParams = (param, value) => {
+  const permalink = new URLSearchParams(window.location.search);
+  if (value) {
+    permalink.set(param, value);
+  } else {
+    permalink.delete(param);
+  }
+  window.history.pushState({}, null, '?' + permalink.toString());
```

Comment:
Do you want to protect from calling it to the same url? To avoid unnecessary recoreds in the history.

Reply:
yes, it was protected but was not working properly with the context, 
I need to think a better way to do it, in "theory" the function should not be called if the filters are the same, in **theory**

Reply:
Exactly :) maybe, at least, we could `console.error` and monitor it in sentry.

Reply:
Tested It manually and works as expected without the protection, maybe the method is smarter than us (?)

Reply:
It's not, but probably your code does :) Seriously, you can call that method with the same URL and it will add a record in the `history` so back button will do nothing.
My concern is for future code that maybe will not be smart enough. We preferred not to bother the user.

Reply:
Guarded 🛡!!!

Reply:
> Tested It manually and works as expected without the protection, maybe the method is smarter than us (?)

I lied, wasn't working as expected, the history was messed up.